File: _src/06/Chapter6.VB/Maps/CompanyMap.vb
Imports System.Data.Entity.ModelConfiguration
Imports Chapter6.VB.Models

Namespace Maps

    Public Class CompanyMap
        Inherits EntityTypeConfiguration(Of Company)
        Public Sub New()
            MapToStoredProcedures( _
                Sub(config)

                    config.Delete(
                        Sub(procConfig)

                        procConfig.HasName("CompanyDelete")
                        procConfig.Parameter(Function(company) company.CompanyId, "companyId")
                    End Sub
                    )
                    config.Insert(Function(procConfig) procConfig.HasName("CompanyInsert"))
                    config.Update(Function(procConfig) procConfig.HasName("CompanyUpdate"))
                End Sub
            )
        End Sub
    End Class
End Namespace

File: _src/06/Chapter6.CSharp/Maps/CompanyMap.cs
using System.Data.Entity.ModelConfiguration;
using Chapter6.CSharp.Models;

namespace Chapter6.CSharp.Maps
{
    public class CompanyMap :
        EntityTypeConfiguration<Company>
    {
        public CompanyMap()
        {
            MapToStoredProcedures(config =>
                {
                    config.Delete(
                        procConfig =>
                        {
                            procConfig.HasName("CompanyDelete");
                            procConfig.Parameter(company => company.CompanyId, "companyId");
                        });
                    config.Insert(procConfig => procConfig.HasName("CompanyInsert"));
                    config.Update(procConfig => procConfig.HasName("CompanyUpdate"));
                });
        }
    }
}
